superset-frontend/plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -226,36 +226,61 @@ const transformProps = (
     emitCrossFilters,
   } = chartProps;
 
+  const [metrics, percentMetrics, columns] = processColumns(chartProps);
+
+  let baseQuery;
+  let countQuery;
+  let totalQuery;
+  let rowCount;
+  if (formData.server_pagination) {
+    [baseQuery, countQuery, totalQuery] = queriesData;
+    rowCount = (countQuery?.data?.[0]?.rowcount as number) ?? 0;
+  } else {
+    [baseQuery, totalQuery] = queriesData;
+    rowCount = baseQuery?.rowcount ?? 0;
+  }
+  const data = processDataRecords(baseQuery?.data, columns);
+
+  if (Number(formData.server_page_length) > 10000) {
```

Comment:
Can we move this to a constant for better visibility?

Reply:
Solved, I created MAX_SERVER_PAGE_LENGTH in consts.ts